app/src/main/java/org/mozilla/focus/observer/LoadTimeObserver.kt

```diff
@@ -0,0 +1,57 @@
+package org.mozilla.focus.observer
+
+import android.os.SystemClock
+import android.support.v4.app.Fragment
+import android.util.Log
+import org.mozilla.focus.architecture.NonNullObserver
+import org.mozilla.focus.session.Session
+import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.utils.UrlUtils
+
+object LoadTimeObserver {
+    private const val MIN_LOAD_TIME: Long = 40
+    private const val MAX_PROGRESS = 99
+    private const val LOG_TAG: String = "LoadTimeObserver"
+
+    @JvmStatic
+    fun addObservers(session: Session, fragment: Fragment) {
+        var startLoadTime: Long = 0
+        var urlLoading: String? = null
+
+        session.loading.observe(fragment, object : NonNullObserver<Boolean>() {
+            public override fun onValueChanged(t: Boolean) {
+                if (t) {
+                    if ((urlLoading != null && urlLoading != session.url.value) || urlLoading == null) {
+                        urlLoading = session.url.value
+                        startLoadTime = SystemClock.elapsedRealtime()
+                        Log.i(LOG_TAG, "zerdatime $startLoadTime - page load $urlLoading start")
+                    }
+                } else {
+                    // Progress of 99 means the page completed loading and wasn't interrupted.
```

Comment:
\[detekt\] INFO: \[style/MagicNumber\] Report magic numbers\. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is\. It's better to declare such numbers as constants and give them a proper name\. By default, \-1, 0, 1, and 2 are not considered to be magic numbers\.